modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractCSharpCodegen.java

```diff
@@ -810,22 +809,6 @@ public void updateCodegenPropertyEnum(CodegenProperty var) {
         }
     }
 
-    /**
-     * Update property if it is a C# value type
-     *
-     * @param models list of all models
-     */
-    protected void updateValueTypeProperty(Map<String, ModelsMap> models) {
-        for (String openAPIName : models.keySet()) {
-            CodegenModel model = ModelUtils.getModelByName(openAPIName, models);
-            if (model != null) {
-                for (CodegenProperty var : model.vars) {
-                    var.vendorExtensions.put("x-is-value-type", isValueType(var));
-                }
-            }
-        }
-    }
-
```

Comment:
If I understand it correctly, x-is-value-type is not identical to x-csharp-value-type

https://github.com/OpenAPITools/openapi-generator/blob/d8ed253439add8c5809ee8f80e88714b19efcdc4/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractCSharpCodegen.java#L834

Are we sure x-is-value-type can be completely removed?

cc @devhl-labs

Reply:
As far as my test went, I can say that `x-is-value-type` was an attempt at dealing with the same thing as `x-csharp-value-type` but it was buggy.
For example, `DateTime` wasn't considered a value type which created lots of warning everywhere that the variable can't be null but we were doing null check on it. 

But to be  honest, I'm still coming back to C# and .NET after almost 8 years in Java so I might be mistaking. I suggest the technical committee takes its time to validate this PR and I'm 100% available to make any changes to it if required!

Reply:
I suspect we should be trying to get rid of x-csharp-value-type and fix whatever is wrong with x-value-type. Or consider if we really even need either. Everything is nullable now, at least with nrt enabled, so we should be using isNullable. Not sure what other purposes either of these extensions serves, but I doubt there are any.

Reply:
Hi @devhl-labs ! Can we see this as work in progress towards this objective? My real goal with all these changes I've done recently is to remove most of the compilation warning in the generated code and most of them come from badly handled null when using nrt. That was just "initial cleanup" I was doing to get there but I agree with you that it might not be still necessary. I suspect that the reason for these types may be in the fact that we support a lot of very old version of .NET and C# in this generator and it might have been to get around some other problem that we can't think of.

So, to merge this PR, should I:
1. Put back the `x-is-value-type` in the `AbstractCSharpCodegen.java` file but not put it back in the mustache files.

OR

2. We consider this a milestone in a bigger cleanup that might end up with `x-csharp-value-type` gone with a better handling of the null value natively with the nrt.

Reply:
@wing328 or @devhl-labs Please consider closing this conversation because if you check my "rebased" version, it was removed in another PR by @devhl-labs. Therefore, it's not part of my changes anymore.